### ISSUE TITLE
no way to test for valid configuration

### ISSUE THREAD

**Issue by alexrudd2** [pomerium/pomerium#3222]
**Is your feature request related to a problem? Please describe.**

Pomerium crashes, hard, on the slightest misconfiguration.  There appears to be no way to check for configuration validity before crashing.  See also #3221

**Describe the solution you'd like**

`nginx -t`
https://adamtheautomator.com/nginx-test-config/

**Describe alternatives you've considered**
I wrote a script to backup old config, restart, check for failure, and restore old configs.  We'd really like to use pomerium's featureset!  We were hoping to have switched over by now, but it's taken a lot of testing :( and there's no way I can deploy without this script.

**Explain any additional use-cases**

-

**Additional context**

docker swarm services allow for pretty fast rollback, fortunately

**Comment by calebdoxsey** [comment]
Agreed that this would be valuable. It's slightly complicated because we have a layered configuration where there can be multiple configuration sources on top of each other. 

The most common is one file/environment source and then a configuration that comes from the databroker.

That said a basic version that tests just the file/environment would be pretty straightforward and would catch a lot of issues.

**Comment by desimone** [comment]
Related
- https://discuss.pomerium.com/t/how-to-test-config-before-restarting-service/216/7
